web/satellite/src/components/referral/ReferralStats.vue

```diff
@@ -23,62 +23,57 @@
-                        }
-                    },
-                    availableCredits: {
-                        title: 'earned credits',
-                        description: 'Free credits that will apply to your upcoming bill',
-                        symbol: '$',
-                        background: {
-                            backgroundColor: 'rgba(217, 225, 236, 0.5)',
-                        }
-                    },
-                    usedCredits: {
-                        title: 'applied credits',
-                        description: 'Free credits that have already been applied to your bill',
-                        symbol: '$',
-                        background: {
-                            backgroundColor: '#D1D7E0',
-                        }
-                    },
-                }
-            };
-        },
-        methods: {
-            fetch: async function() {
-                const creditUsageResponse = await this.$store.dispatch(CREDIT_USAGE_ACTIONS.FETCH);
-                if (!creditUsageResponse.isSuccess) {
-                    this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch credit usage: ' + creditUsageResponse.errorMessage);
-                }
-            }
-        },
-        mounted() {
-            (this as any).fetch();
-        },
-        computed: {
-            title() {
-                let name = this.$store.state.usersModule.fullName || '' ;
-                let text = 'Here Are Your Referrals So Far';
-
-                return name.length > 0 ? `${name}, ${text}` : text;
-            },
-            usage() {
-                return this.$store.state.creditUsageModule.creditUsage;
+    class CreditDescription {
+        public title: string;
+        public description: string;
+        public symbol: string;
+
+        // possibly we could add some 'style' type
+        public style: any;
+
+        constructor(title: string, description: string, symbol: string, color: string) {
+            this.title = title;
+            this.description = description;
+            this.symbol = symbol;
+            this.style = { backgroundColor: color };
+        }
+    }
+
+    @Component
+    export default class ReferralStats extends Vue {
+        private readonly TITLE_SUFFIX: string = 'Here Are Your Referrals So Far';
+        private stats: CreditDescription[] = [
+            new CreditDescription('referrals made', 'People you referred who signed up', '', '#FFFFFF'),
+            new CreditDescription('earned credits', 'Free credits that will apply to your upcoming bill', '$', 'rgba(217, 225, 236, 0.5)'),
+            new CreditDescription('applied credits', 'Free credits that have already been applied to your bill', '$', '#D1D7E0'),
+        ];
+
+        public async mounted() {
+            // TODO: we pre-fetch all data in /src/views/Dashboard, but this is tardigrade related, so could be here
+            this.fetch();
```

Comment:
do we need `await` here so it will wait `fetch` to finish?

Reply:
to prevent screen jitter while data will be fetched